### DISCUSSION TITLE
How to force `readEntity` using Jackson in REST Client Reactive when returned content-type is `text/plain`?

### DISCUSSION THREAD

**Discussion by astappiev** [quarkusio/quarkus#35907] [Q&A]
I'm building a REST Client Reactive for a quite old API endpoint, which supports `json` format, however, it returns an incorrect header - `Content-type: text/plain;charset=utf-8`.

When I'm trying to return an entity, it throws `jakarta.ws.rs.ProcessingException: Response could not be mapped to type class IpernityResponse`.

Is there a way to add custom MessageBodyReader or modify response headers before reading?
Right now, I return `String` from a client and then use `ObjectMapper` to successfully read it.

Thank you in advance

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
You can try playing with a `ReaderInterceptorContext` which you can register as a provider.
